src/test/java/com/unosquare/cvgenerator/serviceimpl/TemplateServiceImplTest.java

```diff
@@ -0,0 +1,91 @@
+package com.unosquare.cvgenerator.serviceimpl;
+
+import com.flextrade.jfixture.JFixture;
+import com.unosquare.cvgenerator.dao.TemplateRepository;
+import com.unosquare.cvgenerator.model.dto.TemplateDetailDTO;
+import com.unosquare.cvgenerator.model.entity.TemplateDetail;
+import com.unosquare.cvgenerator.service.impl.TemplateServiceImpl;
+import com.unosquare.cvgenerator.util.MapperUtil;
+import org.assertj.core.util.Lists;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TemplateServiceImplTest {
+
+    @Mock
+    private MapperUtil mapperUtil;
+
+    @Mock
+    private TemplateRepository repository;
+
+    @InjectMocks
+    private TemplateServiceImpl templateService;
+
+    private JFixture fixture = new JFixture();
+
+
+    @Test
+    public void findAll_whenDataAvailable_returnsDtoList() {
+
+        // arrange
+        List<TemplateDetail> testListTemplateDetail =
+                Lists.newArrayList(fixture.collections().createCollection(TemplateDetail.class));
+
+        fixture.collections().addManyTo(testListTemplateDetail, TemplateDetail.class);
+
+        doReturn(testListTemplateDetail).when(repository).findAll();
+        List<TemplateDetailDTO> testListTemplateDetailDto =
+                mapperUtil.map(testListTemplateDetail, TemplateDetailDTO.class);
+
+        // act
+        List<TemplateDetailDTO> actualListTemplateDetailDto = templateService.findAll();
+
+        // assert
+        Assert.assertEquals(testListTemplateDetailDto, actualListTemplateDetailDto);
+        Mockito.verify(repository, times(1)).findAll();
+    }
+
+    @Test
+    public void findAll_whenNoData_returnsEmptyList() {
+        doReturn(new ArrayList<>()).when(repository).findAll();
```

Comment:
Same as above for style difference. For empty lists, I like to use `emptyList()` static import from `Collections.emptyList()`